bindinfo/cache.go

```diff
@@ -45,6 +52,8 @@ type cache map[string][]*bindMeta
 // Handle holds an atomic cache.
 type Handle struct {
 	atomic.Value
+	lock *sync.Mutex
```

Comment:
How about combine lock and ctx into a single struct as same as `Handle`, `clientConn` and `Tracker`:
```
type Handle struct {
    mu struct {
        sync.Mutex
        ctx sessionCtx.Context
    }
}
```